File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    from collections import defaultdict
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ans = defaultdict(list)

        for word in strs:
            count = 26 * [0]
            # print('word', word)
            for s in word:
                count[ord(s) - ord('a')] += 1
            ans[tuple(count)].append(word)
        
        return list(ans.values())

File: Data Structures & Algorithms/longest-common-prefix/submission-1.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }
        String ans = "";

        int i = 0;
        while (i < strs[0].length()) {
            char lastChar = strs[0].charAt(i);
            for(String s : strs) {
                if (i >= s.length() || lastChar != s.charAt(i)) {
                    return ans;
                }
            }
            ans += lastChar;
            i++;
        }
        return ans;
    }
}